File: admissionService/src/main/java/com/sbgu/admissionservice/demo/Entity/School.java
package com.sbgu.admissionservice.demo.Entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@Entity
@Table(name = "school")
@AllArgsConstructor
@NoArgsConstructor
public class School {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long schoolId;
    private String name;

    @OneToMany(mappedBy = "generatedFor")
    private List<AdmissionRequest> admissionRequests;

    private String website;

    private String location_latitude;
    private String location_longitude;
}

File: UI/school/lib/models/school.dart
class School {
  final String id;
  final String name;
  final String website;
  final String location_longitude;
  final String location_latitude;
  final String address;
  final List<dynamic> eligibilityCriteria;
  final List<dynamic> documentsRequired;

  School(
      {required this.id,
      required this.eligibilityCriteria,
      required this.documentsRequired,
      required this.name,
      required this.website,
      required this.location_latitude,
      required this.address,
      required this.location_longitude});
}
